src/gt4py/stencil_object.py

```diff
@@ -47,7 +58,7 @@ def _compute_cache_key(field_args, parameter_args, domain, origin) -> int:
 
 
 @dataclass(frozen=True)
-class FrozenStencil:
+class FrozenStencil(SDFGConvertible):
```

Comment:
Instead of adding all these attributes directly to `gtscript.StencilObject` and `gtscript.FrozenStencil`, I have two options:

1. I wonder if we could add a wrapper class on top of this that needs to be called if one wants to orchestrate. Something like 
```
obj = gtscript.stencil(backend="gtc:numpy", definition=test_def, **backend_kwargs) # if not using dace orchestration
...
obj = dace_util.wrapper(gtscript.stencil(...)) # if using dace orchestration
```

2. Add these in the module itself generated from the `gtc:dace` backend.

Reply:
3. Made an intermediate level of inheritance that adds the DaCe support.

Reply:
How do you feel about still having the `__sdfg__` in the `StencilObject` just for raising errors? @tbennun  Do you see another way of printing a meaningful error? Currently, if we don't put it there, DaCe will try to parse the StencilObject.

Reply:
so what exactly is happening? is it trying to get `StencilObject.__sdfg__` instead of the inherited class?

Reply:
No it all works fine. Just in the case of non-dace backends, we want to raise a meaningful error, currently we add `__sdfg__` etc to the StencilObject and override them in the dace backends. I was wondering if there was a way to keep the StencilObject free of all things dace. This works fine though, we can keep it.

Reply:
@jdahm you deleted some message I think, do you think they should be NotImplementedError ? To me those usually mean they are not implemented _yet_ but we have no intention of supporting it in the future.

Reply:
Sorry, I deleted it after I thought about it some more. The `TypeError` stubs you have are great!